### PR TITLE
Use solid white background for outline buttons

### PR DESCRIPTION
**Pull request by dimitrisnl**
Fixes: #820
Change-type: minor
Signed-off-by: Dimitrios Lytras <dimitrios@balena.io>

---
##### Contributor checklist

- [ ] I have rebuilt the README with `npm run build:docs`
- [ ] I have regenerated screenshots for any affected components with `npm run generate-screenshots`
- [ ] I have regenerated jest snapshots for any affected components with `npm run jest -- -u`

##### Reviewer Guidelines
- When submitting a review, please pick:
  - '*Approve*' if this change would be acceptable in the codebase (even if there are minor or cosmetic tweaks that could be improved).
  - '*Request Changes*' if this change would not be acceptable in our codebase (e.g. bugs, changes that will make development harder in future, security/performance issues, etc).
  - '*Comment*' if you don't feel you have enough information to decide either way (e.g. if you have major questions, or you don't understand the context of the change sufficiently to fully review yourself, but want to make a comment)
---

### CODE REVIEW

**Comment by dimitrisnl** [comment]
I expect that everyone that checks the Outline button, thinks that the background is of solid white color. In colored backgrounds, the outline button blends in and it's very hard to notice.

File: src/components/Button.tsx [hunk 1]
```diff
@@ -34,6 +34,7 @@ const ColouredButton = styled(ButtonBase)`
 `;
 
 const Outline = styled(ButtonBase)`
+	background: #fff;
```
Comment (sradevski) [inline src/components/Button.tsx]: I would only fall back to `white` if the `bg` prop is not passed to it maybe? Not a strong opinion on this.

Comment (dimitrisnl) [reply src/components/Button.tsx]: I agree @sradevski , that makes sense

**Comment by sradevski** [comment]
@dimitrisnl Just update the snapshots, I see they failed again. Also adding an outlined button to storybook with a bg might be nice.

**Comment by dimitrisnl** [comment]
Screen Shot 2019-05-01 at 10 23 23 AM

That was a great suggestion. Seems like outline button's color, mimics the bg property somehow. 

```js
          <Button
            m={2}
            outline
            bg="#000"
            >Outline</Button>
        </Box>
        <Box m={3}>
          <Button
            m={2}
            primary
            bg='red'
            >Outline</Button>
        </Box>
```

**Comment by sradevski** [comment]
@dimitrisnl when I think about it, shouldn't the outline button actually have transparent background? If you want a button with a white color, you should just use a normal button with a white background.

**Comment by dimitrisnl** [comment]
I don't have any strong opinion, but we don't have any.

What I want, are some simple white background buttons, with green, red, yellow, etc colors to match the products. When using outline they are hidden in the colored background.

**Comment by sradevski** [comment]
@dimitrisnl But in that case, maybe it is better to just use `bg="fff"` on a normal button. Any way is fine, it is just that it feels that an "Outline" button should just have an outline and be clickable, and not have a solid background.

**Comment by dimitrisnl** [comment]
I see. It just seems strange to be that we don't that the simplest form a button, a plain white one.

**Pull request closed without merging**